src/network/wait-online/link.c

```diff
@@ -248,3 +256,47 @@ int link_update_monitor(Link *l) {
 
         return ret;
 }
+
+int link_check_dns_accessible(Link *l, AddressFamily *ret) {
+        AddressFamily accessible = ADDRESS_FAMILY_NO;
+        int r;
+
+        assert(l);
+        assert(ret);
+
+        STRV_FOREACH(d, l->dns_servers) {
+                _cleanup_close_ int fd = -EBADF;
+                union sockaddr_union sa;
+                union in_addr_union in_addr;
+                int family;
+
+                r = in_addr_from_string_auto(*d, &family, &in_addr);
```

Comment:
The string may contain port number and interface specifier.
Please use `in_addr_port_ifindex_name_from_string()`.